pkg/apiserver/profiling/router.go

```diff
@@ -144,6 +144,22 @@ func (s *Service) getGroupDetail(c *gin.Context) {
 		return
 	}
 
+	if taskGroup.State == TaskStateFinish {
+		// fix legacy all finish state for group task
+		finishedTasks := 0
+		for _, task := range tasks {
+			if task.State == TaskStateFinish {
+				finishedTasks++
+			}
+		}
+		if finishedTasks == len(tasks) {
+			taskGroup.State = TaskAllFinish
+		} else {
+			taskGroup.State = TaskPartialFinish
+		}
+		s.params.LocalStore.Save(taskGroup)
+	}
```

Comment:
Is it possible to do modification in other places? It is a bad practice to modify something in a GET, as you will face these problems:

1. Is this GET API still isomorphic?
2. What about concurrent GETs? Will there be concurrent issues?
3. It violates the "GET" restful semantic.

Reply:
Yep, you're right, I have considered writing a script to modify the legacy state, let me think more.

Reply:
If migrating old data is complex then maybe we can just leave it. We can look forward and know that the improvement is still good for new profilings, which is good enough.

Reply:
Yep, it is a little complex, I think it is fine to keep the old data. Let me close this PR and move some refinements in this PR to a new PR with other refinements for conprof.